ref_arch/multi_tenant.rst

```diff
@@ -0,0 +1,540 @@
+
+::
+
+  -- Total campaign budget vs expense this month
+
+  SELECT camp.monthly_budget,
+         CASE cost_model
+         WHEN 'cost_per_click' THEN
+           clicks_count
+         ELSE
+           impressions_count
+         END AS billable_events
+  FROM campaigns camp
+  JOIN ads a ON (
+        a.campaign_id = camp.id
+    AND a.company_id = camp.company_id)
+  WHERE camp.company_id = 5;
+
+Sharing Data Between Tenants
+----------------------------
+
+Up until now all tables have been distributed by :code:`company_id`, but sometimes there is data that can be shared by all tenants, and doesn't "belong" to any tenant in particular. For instance all companies using this example ad platform might want to get geographical information for their audience based on IP addresses. In a single machine database this could be accomplished by a lookup table for geo-ip, like the following. (A real table would probably use PostGIS but bear with the simplified example.)
+
+::
+
+  CREATE TABLE geo_ips (
+    addrs cidr NOT NULL PRIMARY KEY,
+    latlon point NOT NULL
+      CHECK (-90  <= latlon[0] AND latlon[0] <= 90 AND
+             -180 <= latlon[1] AND latlon[1] <= 180)
+  );
+  CREATE INDEX ON geo_ips USING gist (addrs inet_ops);
+
+Joining clicks with this table can produce, for example, the locations of everyone who clicked on ad 456.
+
+::
+
+  SELECT latlon
+    FROM geo_ips, clicks c
+   WHERE addrs >> c.user_ip
+     AND c.clicked_at > current_date - INTERVAL '1 day'
+     AND c.company_id = 5
+     AND c.ad_id = 456;
+
+In Citus we need to find a way to co-locate the :code:`geo_ips` table with clicks for every company. One way would be to add a :code:`company_id` column to :code:`geo_ips` and duplicate the data in the table for every company. This approach is not optimal because it introduces the burden of keeping the data synchronized between the companies if and when it changes. A more convenient way is by designating :code:`geo_ips` as a *reference table.*
+
+Reference tables in Citus have exactly one shard, and it is replicated across all worker nodes. This co-locates the information for all tenants' queries. It does require reserving space for a copy of the data on all nodes, but automatically stays in sync during reference table updates. To make a reference table, call :ref:`create_reference_table <create_reference_table>` for a table on the coordinator node:
+
+::
+
+  -- Make synchronized copies of geo_ips on all workers
+
+  SELECT create_reference_table('geo_ips');
+
+After doing this, the join query (presented earlier) on :code:`geo_ips` and :code:`clicks` will perform efficiently.
+
+Online Changes to the Schema
+----------------------------
+
+Citus propagates most DDL statements fron the coordinator node to the workers. This allows the database administrator to alter the database even during use and after tables are distributed in the cluster. Citus uses a two-phase commit protocol to make sure updates happen consistently.
```

Comment:
Could we add some context / explain the benefit this provides to users? Something like:

Another challenge with multi-tenant systems is ensuring that the schemas for all the tenants are always in sync and schema changes are consistently reflected to all the tenants.

In Citus, you can use standard postgres DDL commands to change the schema of your tables and Citus will propagate them from the coordinator node to the workers.....

i.e. make it more MT focus as opposed to syntax / SQL focus.